src/Catel.Core/Catel.Core.Shared/Data/PropertyDataManager.cs

```diff
@@ -214,6 +242,27 @@ public PropertyData GetPropertyData(Type type, string name)
         }
 
         /// <summary>
+        /// Gets the dynamic properties data.
+        /// </summary>
+        /// <param name="type">The type for which to get the dynamic properties data.</param>
+        /// <returns>The <see cref="IEnumerable{PropertyData}"/> containing dynamic properties data.</returns>
+        public IEnumerable<PropertyData> GetDynamicPropertiesData(Type type)
+        {
+            Argument.IsNotNull("type", type);
+
+            lock (_propertyDataLock)
+            {
+                CatelTypeInfo propertyDataOfType;
+                if (!_propertyData.TryGetValue(type, out propertyDataOfType))
+                {
+                    _propertyData[type] = new CatelTypeInfo(type);
+                }
+
+                return propertyDataOfType.GetCatelProperties().Values.Where(x => x.IsDynamicProperty).ToList();
```

Comment:
Yes. Proper version is:
`if (!_propertyData.TryGetValue(type, out propertyDataOfType))
{
	propertyDataOfType = new CatelTypeInfo(type);
    _propertyData[type] = new CatelTypeInfo(type);
}`
If CatelTypeInfo is immutable or Thread-safe last line can be placed out of lock block